src/pages/Trends/changesEpic.js

```diff
@@ -0,0 +1,136 @@
+      return Observable.from(
+        client.query({
+          query: SOCIAL_VOLUME_QUERY,
+          variables: {
+            word,
+            from: fromDate.toISOString(),
+            to: toDate.toISOString()
+          }
+        })
+      ).map(
+        ({
+          data: { telegram, reddit, discord, professional_traders_chat: ptc }
+        }) => {
+          const [
+            oldValue = defaultMentionsCount,
+            newValue = defaultMentionsCount
+          ] = mergeTimeseriesByKey({
+            timeseries: [
+              telegram.chartData,
+              reddit.chartData,
+              discord.chartData,
+              ptc.chartData
+            ],
+            mergeData: (longestTSData, timeserieData) => {
+              return {
+                mentionsCount:
+                  longestTSData.mentionsCount + timeserieData.mentionsCount,
+                datetime: longestTSData.datetime
+              }
+            }
+          }).slice(-2)
+
+          return {
+            [word]: [oldValue.mentionsCount, newValue.mentionsCount]
+          }
+        }
+      )
+    })
+
+    return Observable.forkJoin(...words)
+      .flatMap(result => {
+        return Observable.of({
+          type: TREND_WORD_VOLUME_CHANGE_FULFILLED,
+          payload: result.reduce((acc, value) => {
+            Object.assign(acc, value)
+            return acc
+          }, {})
+        })
+      })
+      .catch(handleErrorAndTriggerAction('changes failed'))
+  })
+
+export const wordTrendScoreEpic = (action$, store, { client }) =>
+  action$.ofType('[trends] HYPED_FETCH_SUCCESS').mergeMap(({ payload }) => {
+    // HACK(vanguard): wordTrendScore from/to does not work correctly
+    // Can't fetch only needed time period, should fetch for all day
+    const fromDate = new Date()
+    fromDate.setHours(0, 0, 0, 0)
+    fromDate.setDate(fromDate.getDate() - 2)
+    const toDate = new Date()
+    toDate.setHours(24, 0, 0, 0)
+
+    const { items } = store.getState().hypedTrends
+    const { topWords } = items[items.length - 1]
+
+    const words = topWords.map(({ word }) => {
+      return Observable.from(
+        client.query({
+          query: WORD_TREND_SCORE_QUERY,
+          variables: {
+            word,
+            from: fromDate,
+            to: toDate
+          }
+        })
+      ).map(({ data: { wordTrendScore } }) => {
+        const sorted = wordTrendScore.sort(
+          ({ datetime: aDate }, { datetime: bDate }) =>
+            new Date(aDate) < new Date(bDate) ? 1 : -1
+        )
+        const [{ score: scoreNew }, { score: scoreOld }] = sorted
+        return {
+          [word]: [parseInt(scoreOld, 10), parseInt(scoreNew, 10)]
+        }
+      })
+    })
+
+    return Observable.forkJoin(...words)
+      .flatMap(result => {
+        return Observable.of({
+          type: TREND_WORD_SCORE_CHANGE_FULFILLED,
+          payload: result.reduce((acc, value) => {
+            Object.assign(acc, value)
```

Comment:
@DmitriVanGuard Why not spread operator. Node.js team said: "The object rest/spread operators are both syntactically neat and offer performance benefits over Object.assign()"
Sure it's a browser, but I don't think you win a lot here. If you don't have much better performance, better to use Spread. And for safety, better to use Object.assign({}, acc, value)

Reply:
1. It's an array loop where the performance matter;
2. No need for `immutability` (Object.assign({}, acc, value)), because the `acc` is used only in `reduce` context;
3. `Object.assign()` is standartized, doesn't require pollyfils and without extra allocations is significantly faster.
Benchmark: https://jsperf.com/object-assign-vs-spread
![image](https://user-images.githubusercontent.com/25135650/55402740-d9be7280-555c-11e9-858f-ffced3c02df6.png)